helper/bearer.go

```diff
@@ -25,9 +25,28 @@ import (
 	"strings"
 )
 
-func BearerTokenFromRequest(r *http.Request) string {
-	auth := r.Header.Get("Authorization")
-	split := strings.SplitN(auth, " ", 2)
+const (
+	defaultAuthorizationHeader = "Authorization"
+)
+
+type BearerTokenLocation struct {
+	Header         *string `json:"header"`
+	QueryParameter *string `json:"query_parameter"`
+}
+
+func BearerTokenFromRequest(r *http.Request, tokenLocation *BearerTokenLocation) string {
+	var token string
+	if tokenLocation != nil {
+		if tokenLocation.Header != nil {
+			token = r.Header.Get(*tokenLocation.Header)
+		} else if tokenLocation.QueryParameter != nil {
+			token = r.FormValue(*tokenLocation.QueryParameter)
+		}
```

Comment:
Is it possible that both if / else if do not get executed? Then `token` would be empty.

Reply:
That is not possible, as JSON schema for `jwt` and `oauth2_introspection` authenticators would not allow for that.

Reply:
Can we make this explicit in the code by simply doing a if/else?

Reply:
We can, but if somehow QueryParameter would be a nil, than the program will crash on dereferencing a nil pointer. With "else if" the function will simply return an empty string, like if there was no token in the configured header/query parameter (what may be a proper behavior if there was "alternative token location" set, but not configured - even though with the current schema it is not possible).

I see an alternative solution here: returning an error saying that rule  is misconfigured. But I would have to add error handling in the places where this function is called. 

What do you think?

Reply:
Ok, how about simply falling back to the authorization header in the case where both are nil?

Reply:
Ok, done